File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/service/AdminAssetCategoryService.java
package com.NBE3_4_2_Team4.domain.asset.main.service;

import com.NBE3_4_2_Team4.domain.asset.main.dto.AdminAssetCategoryRes;
import com.NBE3_4_2_Team4.domain.asset.main.entity.AdminAssetCategory;
import com.NBE3_4_2_Team4.domain.asset.main.repository.AdminAssetCategoryRepository;
import com.NBE3_4_2_Team4.global.exceptions.ServiceException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class AdminAssetCategoryService {

    private final AdminAssetCategoryRepository adminAssetCategoryRepository;

    @Transactional
    public AdminAssetCategory create(String name) {
        if (adminAssetCategoryRepository.existsByName(name))
                throw new ServiceException("404-1", "이미 존재하는 이름입니다");

        AdminAssetCategory adminAssetCategory = new AdminAssetCategory(name);
        return adminAssetCategoryRepository.save(adminAssetCategory);
    }

    @Transactional(readOnly = true)
    public List<AdminAssetCategoryRes> findAll() {
        return adminAssetCategoryRepository.findAllByIsDisabledFalse()
                .stream()
                .map(AdminAssetCategoryRes::from)
                .toList();
    }

    @Transactional
    public AdminAssetCategory updateName(long id, String newName) {
        AdminAssetCategory adminAssetCategory = adminAssetCategoryRepository.findById(id)
                .orElseThrow(() -> new ServiceException("404-1", "존재하지 않는 id 입니다"));
        adminAssetCategory.setName(newName);
        return adminAssetCategoryRepository.save(adminAssetCategory);
    }

    @Transactional
    public AdminAssetCategory disable(long id) {
        AdminAssetCategory adminAssetCategory = adminAssetCategoryRepository.findById(id)
                .orElseThrow(() -> new ServiceException("404-1", "존재하지 않는 id 입니다"));
        adminAssetCategory.setDisabled(true);
        return adminAssetCategoryRepository.save(adminAssetCategory);
    }

    public long count() {
        return adminAssetCategoryRepository.count();
    }
}

File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/service/AdminAssetCategoryService.kt
package com.NBE3_4_2_Team4.domain.asset.main.service

import com.NBE3_4_2_Team4.domain.asset.main.dto.AdminAssetCategoryRes
import com.NBE3_4_2_Team4.domain.asset.main.entity.AdminAssetCategory
import com.NBE3_4_2_Team4.domain.asset.main.repository.AdminAssetCategoryRepository
import com.NBE3_4_2_Team4.global.exceptions.ServiceException
import lombok.RequiredArgsConstructor
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class AdminAssetCategoryService (
    private val adminAssetCategoryRepository: AdminAssetCategoryRepository
) {

    @Transactional
    fun create(name: String): AdminAssetCategory {
        if (adminAssetCategoryRepository.existsByName(name)) throw ServiceException("404-1", "이미 존재하는 이름입니다")

        val adminAssetCategory = AdminAssetCategory(name)
        return adminAssetCategoryRepository.save(adminAssetCategory)
    }

    @Transactional(readOnly = true)
    fun findAll(): List<AdminAssetCategoryRes> {
        return adminAssetCategoryRepository.findAllByIsDisabledFalse()
            .map(AdminAssetCategoryRes::from)
    }

    @Transactional
    fun updateName(id: Long, newName: String): AdminAssetCategory {
        val adminAssetCategory = adminAssetCategoryRepository.findByIdOrNull(id)
            ?: throw ServiceException("404-1", "존재하지 않는 id 입니다")

        adminAssetCategory.name = newName
        return adminAssetCategoryRepository.save(adminAssetCategory)
    }

    @Transactional
    fun disable(id: Long): AdminAssetCategory {
        val adminAssetCategory = adminAssetCategoryRepository.findByIdOrNull(id)
            ?: throw ServiceException("404-1", "존재하지 않는 id 입니다")

        adminAssetCategory.isDisabled = true
        return adminAssetCategoryRepository.save(adminAssetCategory)
    }

    fun count(): Long {
        return adminAssetCategoryRepository.count()
    }
}
